torch/csrc/api/include/torch/nn/functional/activation.h

```diff
@@ -40,6 +40,35 @@ inline Tensor logsigmoid(const Tensor& input) {
   return torch::log_sigmoid(input);
 }
 
+inline int _get_softmax_dim(const std::string& name, int ndim) {
+  TORCH_WARN("Implicit dimension choice for ", name, " has been deprecated. "
+             "Change the call to include dim=X as an argument.");
+  int ret;
+  if (ndim == 0 || ndim == 1 || ndim == 3) {
+    ret = 0;
+  } else {
+    ret = 1;
+  }
+  return ret;
+}
```

Comment:
In the Python version, this function is for maintaining backward compatibility for the use cases where people don't pass the `dim` argument to `torch.nn.functional.softmax`. Since `torch::nn::functional::softmax` in C++ API is a new addition, we don't need to maintain any backward compatibility, and we should remove this function and make `dim` a non-optional argument to match the newest Python API design.

Reply:
I understood the background and deleted the function `_get_softmax_dim`.